packages/smooth_app/lib/database/dao_product_list.dart

```diff
@@ -372,6 +373,19 @@ class DaoProductList extends AbstractDao {
     );
   }
 
+  /// Clears a product list: unlinks all its items
+  Future<void> clear(final ProductList productList) async =>
+      localDatabase.database.transaction(
+        (final Transaction transaction) async {
+          final int? id = await _getId(productList, transaction);
```

Comment:
I'am cusious, what happens when _getId retuns null

Reply:
@M123-dev You can find the answer in `DaoProductExtra._getExtraKey`. If I sum up, some lists will always be there and are always meaningful and unique (e.g. history, scan history). They don't need an id: there's only one scan history, we can give it the id `'scanHistory'`. Other lists are user-defined: you can create them and delete them. They need a specific id, like `list/$id`.